Extra optuna

OfficiallyDAC opened the pull request:
Pull request for branch _extra_optuna_ created starting from branch _exercise1_git_. The experiment concerns the usage of Optuna and Hydra Optuna Sweeper for hyperparameters management and optimisation.

sscardapane commented:
Nice! A few comments before merging:

1. It would be better to start the exercise from `exercise2_hydra` so that step 1 is not needed. You can specify that `exercise1_git` is required, similarly to here: [https://github.com/sscardapane/reprodl2021/tree/exercise4_docker#prerequisites](https://github.com/sscardapane/reprodl2021/tree/exercise4_docker#prerequisites).

2. Is the numbering in the README correct? There is nothing after "essentially constituted by 3 major blocks", or "Load and wrap data using dataloaders", and afterwards the numbering starts again from 3.

I am moving this to draft. If point (1) is okay I can create a new branch from there where you can merge afterwards.

OfficiallyDAC commented:
Thanks for the comments! 
As far as point 2 is concerned, there was a mistake. I fixed it. 
Regarding point 1, I agree: I have already modified README.md file as you suggested.

sscardapane commented:
@OfficiallyDAC I have created a new branch for merging starting from `exercise1_git_completed`, but there are a couple of conflicts. Can you solve them? I'll merge afterwards.

OfficiallyDAC commented:
Branch moved to exercise1_git_completed and made a new pull request onto extra_optuna
